### PR TITLE
replace README with Contribution Guide

### PR DESCRIPTION
**Pull request by Brilator**
Idea: let the contribution guide render as html, but "hidden" (not in sidebar). Might be more appealing than just GitHub.  

This way we also really have just **one** README for the repo, where we can link to for installation, testing, etc.

### CODE REVIEW

File: src/docs/ContributionGuide.md [hunk 1]
```diff
@@ -1,38 +1,55 @@
-# How-to "knowledge base"
+---
+layout: docs
```
Comment (Freymaurer) [inline src/docs/ContributionGuide.md:2]: I must say i do not really agree with this decision. If the change is more than just fixing a typo the change author should actually use an markdown viewer or be knowledgeable enough about markdown to use the README.md, which also has the added advantage to be displayed in GitHub if you navigate to the [docs](https://github.com/nfdi4plants/nfdi4plants.knowledgebase/tree/main/src/docs) folder and to keep and unified naming.

In addition i would advice against using README.md and ContributionGuide.md, as this would create maintenance issues.

**What we CAN do is:**
Start parsing README.md, so we also have an html. In this case we would need to add the docs layout metadata to the README.md but otherwise it would be the same.

**Review by Freymaurer — CHANGES_REQUESTED** [review]

File: src/docs/ContributionGuide.md (hunk 1 above)
Comment (Brilator) [reply src/docs/ContributionGuide.md]: > Start parsing README.md, so we also have an html. In this case we would need to add the docs layout metadata to the README.md but otherwise it would be the same.

Yes! Let's do that.

Comment (Brilator) [reply src/docs/ContributionGuide.md]: For ignored stuff we have the _ignored folder. Every other md should be rendered (and then requires yaml)

File: src/docs/README.md [hunk 2]
```diff
@@ -41,18 +41,34 @@ For a general introduction to writing markdown, see: [Markdown tutorial](tutoria
...
+## Common errors
+
+- Missing metadata block
+- Missing or false required (`MUST`) attribute in metadata block
+- Using a layout in metadata block that does not exist
+- Wrong links
+  - to sidebar elements
+  - to images
+
+## Reviewing
```
Comment (Freymaurer) [inline src/docs/README.md:67]: Empty header

File: src/docs/README.md [hunk 3]
```diff
@@ -119,13 +135,16 @@ To add a sidebar element to the page, use the codeblock syntax:
   - Only headers up to `###` are parsed. All header with more depth are parsed to `###`.
 - Tries to match active browser url to referenced ``href`` of any element to set active page.
 
-## _ignored
 
-Anything inside this folder will not be used to generate html pages.
+## Public or not public
```
Comment (Freymaurer) [inline src/docs/README.md:134]: Please rephrase these changes, not descriptive enough. 

For example i do not understand what the following is meant to confer.
```markdown
### visible or hidden

- sidebar
```

File: src/docs/README.md [hunk 4]
```diff
@@ -41,18 +41,34 @@ For a general introduction to writing markdown, see: [Markdown tutorial](tutoria
...
 4. Once you want to submit those changes to the main repository, you can open a "pull request" by clicking "Contribute" (5) in the top-right corner.
    > Remember to "Fetch upstream" (4), if your fork is not up-to-date with the main repository.
 
 5. If edited or added existing content, please assign the original author during your pull-request to review your changes.
+  
   - The github user name of the original author is stored in the yaml block on top under `author_github`
   - During the pull-request mention the author via `@<author_github>`.
 
-# (Live) Testing
+## Testing
```
Comment (Freymaurer) [inline src/docs/README.md:50]: Update toc if you change header level here

File: src/docs/README.md [hunk 5]
```diff
@@ -137,31 +156,53 @@ Any file named `README.md` (case sensitive!) will not be used to generate html p
 
 ## Structure and Format
 
-- Max. 3 headline levels (## & ###)
+Try to avoid deep structures by using no more than two headline levels, i.e.
```
Comment (Freymaurer) [inline src/docs/README.md:160]: Not sure if we need to enforce this, but that is fine with me if you want this 👍

File: src/docs/README.md [hunk 6]
```diff
@@ -137,31 +156,53 @@ Any file named `README.md` (case sensitive!) will not be used to generate html p
...
 
 ## File Naming
 
-Preferred: 
-- [snake_case](https://en.wikipedia.org/wiki/Snake_case) (lower_case_with_underscores) 
+Avoid
+
+- special characters
+- spaces
+- abbreviations
```
Comment (Freymaurer) [inline src/docs/README.md:179]: I think this case is better conveyed by

```markdown
For easier findability, use the article's title for the filename, e.g.:
  DataManagementPlan.md -> `title: Data Management Plan`
```

File: src/docs/README.md [hunk 7]
```diff
@@ -137,31 +156,53 @@ Any file named `README.md` (case sensitive!) will not be used to generate html p
...
+ 
 ## Images
 
 - simple markdown logic (not HTML): `![name_of_image](path_to_image.png)`
 
 ## File Naming
 
-Preferred: 
-- [snake_case](https://en.wikipedia.org/wiki/Snake_case) (lower_case_with_underscores) 
+Avoid
```
Comment (Freymaurer) [inline src/docs/README.md:175]: Rephrase this to `File Name Requirements` and follow requirement level naming: https://www.ietf.org/rfc/rfc2119.txt

File: src/docs/README.md [hunk 8]
```diff
@@ -137,31 +156,53 @@ Any file named `README.md` (case sensitive!) will not be used to generate html p
...
+
+- special characters
+- spaces
```
Comment (Freymaurer) [inline src/docs/README.md:178]: Very good point! Totally forgot about this!

File: src/docs/README.md (hunk 2 above)
Comment (Brilator) [reply src/docs/README.md:67]: That's a place holder. Removed for now.

File: src/docs/README.md (hunk 4 above)
Comment (Brilator) [reply src/docs/README.md:50]: TOC is already removed / added to yaml.

File: src/docs/ContributionGuide.md (hunk 1 above)
Comment (Freymaurer) [reply src/docs/ContributionGuide.md]: Will need to implement the README.md parsing, but thats no issue. Just so you know it won't be available out of the box.

File: src/docs/README.md (hunk 4 above)
Comment (Freymaurer) [reply src/docs/README.md:50]: Actually please revert to manual toc. I know this is cumbersome, but the nfdi-toc is not generated in GitHub.

Comment (Brilator) [reply src/docs/README.md:50]: Ah good point.

File: src/docs/README.md (hunk 3 above)
Comment (Freymaurer) [reply src/docs/README.md:152]: Change: 
- `Public or not public` to `Markdown to HTML Rendering Rules`
- Captitalize non-filler words in headers for consistency:
  - Ignored **C**ontent
  - **V**isible or **H**idden

File: src/docs/README.md (hunk 6 above)
Comment (Freymaurer) [reply src/docs/README.md:195]: Not resolved, my suggestion ist:

```markdown
File names:
- MUST use the article's title for the filename, e.g.: DataManagementPlan.md -> `title: Data Management Plan`
- MUST NOT contain special characters
- MUST NOT contain spaces
- MUST use [snake_case](https://en.wikipedia.org/wiki/Snake_case) (lower_case_with_underscores) OR [PascalCase](https://techterms.com/definition/pascalcase) (UpperCase)
```

Comment (Brilator) [reply src/docs/README.md:195]: Looks good, add it :)

**Review by Freymaurer — APPROVED** [review]
Will merge asap 👍
